File: Other OJ Problems/UVa/p10127 - Ones.cpp
//UVa p10127 - Ones

#include <bits/stdc++.h>

using namespace std;

typedef __int128 lli;

int main(){
	long long int n;
	while(cin >> n){
		lli ac = 0;
		int ans = -1;
		if(n == 0){
			cout << "0" << endl;
			continue;
		}

		for(lli i=1, k=1; i<1e18; i *= 10, k++){
			ac += i;
			if(ac%n == 0){
				ans = k;
				break;
			}
		}
		
		if(ans > 0)
			cout << ans << endl;
	}
}

File: Other OJ Problems/UVa/p10127 - Ones.py
#UVa p10127 - Ones

import sys

for line in sys.stdin:
	for n in line.split():
		n = int(n)

		suma = int(0)
		ans = int(-1)

		if n == 0:
			print("0")
			continue

		i = int(1)
		k = int(1)
		while ans < 0:
			suma = suma + i

			if suma % n == 0:
				ans = k
				break

			i = i*10
			k = k+1

		if ans > 0:
			print(ans) 
